### PR TITLE
nixos-render-docs: take header and footer on CLI

### PR DESCRIPTION
**Pull request by considerate**
## Description of changes

This PR generalizes nixos-render-docs for manpage generation to take the header preamble and footer postscript as file paths as CLI flags --header and --footer.

The driving use-case for this is to support generation of the `home-manager` manpages https://github.com/nix-community/home-manager/pull/4673 using this package now that `optionsDocBook` has been deprecated.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### Priorities

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by pennae — CHANGES_REQUESTED** [review]
this looks pretty reasonable. thanks for taking care of home-manager!

the nixos manpages render the same before and after. for some reason mypy type checks have started to fail at some point in completely unrelated places, no clue what that's about :/

File: pkgs/tools/nix/nixos-render-docs/src/nixos_render_docs/options.py [hunk 1]
```diff
@@ -287,18 +288,27 @@ class ManpageConverter(BaseConverter[OptionsManpageRenderer]):
     _links_in_last_description: Optional[list[str]] = None
 
     def __init__(self, revision: str,
+                 header: Path | None,
+                 footer: Path | None,
```
[the remark this answers is not in the document]
Comment (considerate) [reply pkgs/tools/nix/nixos-render-docs/src/nixos_render_docs/options.py:292]: Ok, that makes sense to me. I'll update add a commit updating this behavior.

**Comment by considerate** [comment]
> for some reason mypy type checks have started to fail at some point in completely unrelated places, no clue what that's about :/

I got some type errors due to nested lookups on unimported modules, is this what you're getting as well?

For example:
```python
    def heading_ids(state: markdown_it.rules_core.StateCore) -> None:
```
breaks on `rules_core` not a known member of `markdown_it`, because the `markdown_it` doesn't declare an attribute named `rules_core`, it's a nested module. To solve this I have to `import markdown_it.rules_core` as well.

**Comment by considerate** [comment]
@pennae I should have mentioned that I used `pyright`, it intentionally errors on those kind of usages apparently:
https://microsoft.github.io/pyright/#/import-statements?id=unsupported-loader-side-effects

**Pull request closed without merging**